coffee-chats/src/main/webapp/src/pages/MainPage.tsx

```diff
@@ -1,52 +1,112 @@
-import React, { useState } from "react";
-import {Box, Button, Container, Grid, Icon, IconButton, InputAdornment,
-  TextField, Tooltip, Typography} from "@material-ui/core";
+import React, { useState, ChangeEvent, useEffect } from "react";
+import {Box, Container, Grid, Icon, IconButton, TextField, Tooltip, Typography,
+  Tabs, Tab, Chip} from "@material-ui/core";
+import { Autocomplete, createFilterOptions } from '@material-ui/lab'
 import {ConnectBackCard} from "../components/ConnectBackCard";
 import { FindChatCard } from "../components/FindChatCard";
+import { capitaliseEachWord } from "../util/stringUtils";
+import { fetchTags } from "../requests/tags";
+
 
 export function MainPage() {
-  const [searchTerms, setSearchTerms] = useState<string[]>([]);
+  const CHATS_VIEW = 0;
+  
+  const filter = createFilterOptions<string>();
+  const [tagOptions, setTagOptions] = useState<string[]>([]);
+  const [tags, setTags] = useState<string[]>([]);
+
+  const [currView, setCurrView] = useState(CHATS_VIEW);
+
+  useEffect(() => {
+    const initFetchTags = (async () => {
```

Comment:
Why do you need so many nested lambdas?
What about this?

```js
useEffect(async () => {
  const fetchedTags = await fetchTags();
  setTagOptions(fetchedTags as string[]);
}, []);
```

Reply:
Unfortunately this is due to the typing within Typescript not matching. The async function is of type () => Promise<void>, and use effect expects a parameter of type void | () => void | undefined. So there has to be an invocation of the function to get around this.

See [this](https://medium.com/javascript-in-plain-english/how-to-use-async-function-in-react-hook-useeffect-typescript-js-6204a788a435) for a further explanation.

Reply:
Interesting, thanks